### ISSUE TITLE
Wrong output

### ISSUE THREAD

**Issue by Karmik25** [ankandrew/fast-plate-ocr#85]
After carefully going through https://github.com/ankandrew/fast-plate-ocr/blob/master/examples/tutorial_fine_tune_plate_model.ipynb, i trained the model and got 
Epoch 30: val_acc did not improve from 0.92292
24/24 ━━━━━━━━━━━━━━━━━━━━ 1s 41ms/step - acc: 0.9815 - char_acc: 0.9989 - len_acc: 0.9935 - loss: 0.0055 - top3_acc: 0.9998 - val_acc: 0.9229 - val_char_acc: 0.9936 - val_len_acc: 0.9937 - val_loss: 0.0272 - val_top3_acc: 0.9969

Now I have my .keras model, and we want to convert the model to .onnx. using the commands given in the notebook, i was not able to convert it to onnx due to some Erfc issue, and with the help of llms i tried to fix it. The commands were

Step 1 — Export Keras → SavedModel
pythonpython -c "
import fast_plate_ocr.train.model.layers
import keras
model = keras.models.load_model(
    '/media/8tb_drive/karmik/fastOCR/trained_models/2026-03-25_12-19-47/best.keras',
    compile=False
)
model.export('/tmp/best_saved_model')
"

Step 2 — Convert SavedModel → ONNX (opset 13)
bashpython -m tf2onnx.convert \
  --saved-model /tmp/best_saved_model \
  --opset 13 \
  --output /media/8tb_drive/karmik/fastOCR/trained_models/2026-03-25_12-19-47/best.onnx

Step 3 — Patch Erfc nodes
python patch.py \
  /media/8tb_drive/karmik/fastOCR/trained_models/2026-03-25_12-19-47/best.onnx \
  /media/8tb_drive/karmik/fastOCR/trained_models/2026-03-25_12-19-47/best_patched.onnx

where patch.py is 
import onnx
import numpy as np
from onnx import helper, TensorProto, numpy_helper

def patch_erfc_gelu(input_path: str, output_path: str):
    model = onnx.load(input_path)
    graph = model.graph
    nodes = list(graph.node)

    erfc_nodes = {n.name: n for n in nodes if n.op_type == "Erfc"}
    if not erfc_nodes:
        print("No Erfc nodes found — model is already clean.")
        onnx.save(model, output_path)
        return

    print(f"Found {len(erfc_nodes)} Erfc node(s). Patching...")

    initializers = list(graph.initializer)
    added_consts = {}

    def add_const(name, value):
        if name not in added_consts:
            tensor = numpy_helper.from_array(np.array(value, dtype=np.float32), name=name)
            initializers.append(tensor)
            added_consts[name] = True

    SQRT_2_OVER_PI = np.float32(np.sqrt(2.0 / np.pi))
    COEFF          = np.float32(0.044715)
    ONE            = np.float32(1.0)

    new_nodes = []
    for node in nodes:
        if node.op_type != "Erfc":
            new_nodes.append(node)
            continue

        erfc_in  = node.input[0]
        erfc_out = node.output[0]
        suffix   = node.name.replace("/", "_").replace(":", "_")

        sqrt2_name   = f"_patch_sqrt2_{suffix}"
        coeff_name   = f"_patch_coeff_{suffix}"
        sqrt2pi_name = f"_patch_sqrt2pi_{suffix}"
        one_name     = f"_patch_one_{suffix}"

        add_const(sqrt2_name,   np.sqrt(2.0))
        add_const(coeff_name,   COEFF)
        add_const(sqrt2pi_name, SQRT_2_OVER_PI)
        add_const(one_name,     ONE)

        x_name    = f"_patch_x_{suffix}"
        x2_name   = f"_patch_x2_{suffix}"
        x3_name   = f"_patch_x3_{suffix}"
        cx3_name  = f"_patch_cx3_{suffix}"
        sum_name  = f"_patch_sum_{suffix}"
        sc_name   = f"_patch_sc_{suffix}"
        tanh_name = f"_patch_tanh_{suffix}"
        t1_name   = f"_patch_t1_{suffix}"

        new_nodes += [
            helper.make_node("Mul",      [erfc_in, sqrt2_name],           [x_name]),
            helper.make_node("Mul",      [x_name, x_name],                [x2_name]),
            helper.make_node("Mul",      [x2_name, x_name],               [x3_name]),
            helper.make_node("Mul",      [x3_name, coeff_name],           [cx3_name]),
            helper.make_node("Add",      [x_name, cx3_name],              [sum_name]),
            helper.make_node("Mul",      [sum_name, sqrt2pi_name],        [sc_name]),
            helper.make_node("Tanh",     [sc_name],                       [tanh_name]),
            helper.make_node("Add",      [one_name, tanh_name],           [t1_name]),
            helper.make_node("Identity", [t1_name],                       [erfc_out]),
        ]

    new_graph = helper.make_graph(
        new_nodes, graph.name,
        list(graph.input), list(graph.output), initializers,
    )
    new_model = helper.make_model(new_graph, opset_imports=model.opset_import)
    new_model.ir_version = model.ir_version
    onnx.checker.check_model(new_model)
    onnx.save(new_model, output_path)
    print(f"Patched model saved → {output_path}")

if __name__ == "__main__":
    import sys
    src = sys.argv[1] if len(sys.argv) > 1 else "best.onnx"
    dst = sys.argv[2] if len(sys.argv) > 2 else "best_patched.onnx"
    patch_erfc_gelu(src, dst)

Step 4 — Fix input name + add Cast (uint8→float32)
pythonpython -c "
import onnx
from onnx import helper, TensorProto

model = onnx.load('/media/8tb_drive/karmik/fastOCR/trained_models/2026-03-25_12-19-47/best_patched.onnx')
graph = model.graph

for node in graph.node:
    for i, inp in enumerate(node.input):
        if inp == 'input_layer':
            node.input[i] = '_cast_out'

graph.input[0].name = 'input'
graph.input[0].type.tensor_type.elem_type = TensorProto.UINT8

cast_node = helper.make_node('Cast', ['input'], ['_cast_out'], to=TensorProto.FLOAT)
graph.node.insert(0, cast_node)

onnx.checker.check_model(model)
onnx.save(model, '/media/8tb_drive/karmik/fastOCR/trained_models/2026-03-25_12-19-47/best_patched.onnx')
print('Done.')
"

now after running the inference, i get ALPRResult(detection=DetectionResult(label='License Plate', confidence=0.8491254448890686, bounding_box=BoundingBox(x1=1496, y1=713, x2=1695, y2=846)), ocr=OcrResult(text='VVVU88VRVRRRRRVVV_RR', confidence=[0.9999850988388062, 0.9999967813491821, 0.9999983310699463, 1.0, 1.0, 0.9955950379371643, 0.9999715089797974, 1.0, 0.9999977350234985, 0.9999997615814209, 1.0, 1.0, 1.0, 1.0, 0.9999673366546631, 0.9839365482330322, 1.0, 1.0, 1.0, 1.0], region=None, region_confidence=None))

**Comment by ankandrew** [comment]
Next time put that code in code snippet to make it easy to follow ... Regarding issue, use `v2` models which use `silu` and don't have those problems, the example you mentioned still uses `v1`, btw your `plate_acc` could be higher (check your data is properly labeled). If you want to export what you have already trained, do the following:

1. Git clone and check following tag https://github.com/ankandrew/fast-plate-ocr/releases/tag/v1.0.2
2. Do `make install` (make sure you have uv package manager)
3. That will create `.venv`, to activate you can also use uv and run export script like the following:
```shell
uv run python -m fast_plate_ocr.cli.export -m /path/to/best.keras --plate-config-file /path/to/plate_config.yaml
```
